src/Illuminate/Validation/ConditionalRules.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Illuminate\Validation;
+
+class ConditionalRules
+{
+    /**
+     * The boolean condition indicating if the rules should be added to the attribute.
+     *
+     * @var callable|bool
+     */
+    protected $condition;
+
+    /**
+     * The rules to be added to the attribute.
+     *
+     * @var array
+     */
+    protected $rules;
+
+    /**
+     * Create a new conditional rules instance.
+     *
+     * @param  callable|bool  $condition
+     * @param  array|string  $rules
+     * @return void
+     */
+    public function __construct($condition, $rules)
+    {
+        $this->condition = $condition;
+        $this->rules = $rules;
+    }
+
+    /**
+     * Determine if the conditional rules should be added.
+     *
+     * @param  array  $data
+     * @return bool
+     */
+    public function passes(array $data = [])
+    {
+        return is_callable($this->condition)
+                    ? call_user_func($this->condition, $data)
```

Comment:
Wrapping `$data` in `Illuminate\Support\Fluent` would make accessing the inputs slightly cleaner (`$inputs->key`) through the `__get` helper that checks if the property exists and otherwise return null, as opposed to needing to use `Arr::get($inputs, 'key')` or `$inputs['key'] ?? null`.

```suggestion
                    ? call_user_func($this->condition, new Fluent($data))
```

Not sure if it's best to put it here or in the Validator on line 1085. Applied here, the setters would have no side-effects on the other conditionals.